### PR TITLE
Implement Chef::Resource#copy_properties_from

### LINKED ISSUE #7046: Port copy_properties_to from etcd cookbook into core chef
```ruby
def copy_properties_to(to, *properties)
  properties = self.class.properties.keys if properties.empty?
  properties.each do |p|
    # If the property is set on from, and exists on to, set the
    # property on to
    if to.class.properties.include?(p) && property_is_set?(p)
      to.send(p, send(p))
    end
  end
end
```

I think it would be better to have it shaped more like:

```ruby
declare_resource(:remote_file, name, &block).apply_properties(include: [ :group, :owner ], exclude: [ :headers ])
```

Maybe `include` is the obvious enough default to have it be `def apply_properties(*include, exclude: [])`

Kind of think it should hang off of Chef::Resource so it method chains better like that.  And rename it to not conflict with existing cookbooks.   Could also write a minimum viable compat_apply_properties cookbook to patch up old chef-client versions so other cookbooks could use it (not the entire crazysauce of compat_resource, but do-just-one-small-thing-only).

### PR DESCRIPTION
**Pull request by lamont-granquist**
Kept in the properties mixin since it is tightly coupled.

closes #7046

### CODE REVIEW

File: lib/chef/mixin/properties.rb [hunk 1]
```diff
@@ -315,6 +315,41 @@ def property_description(name)
...
+      #   r = declare_resource(:file, "etc/whatever.xyz")
+      #   r.copy_properties_from(new_resource, :owner, :group, :mode)
+      # end
+      #
+      # @param other [Object] the other object (Chef::Resource) which implements the properties API
+      # @param includes [Array<Symbol>] splat-args list of symbols of the properties to copy.
+      # @param exclude [Array<Symbol>] list of symbosl of the properties to exclude.
+      # @return the self object the properties were copied to for method chaining
+      #
+      def copy_properties_from(other, *includes, exclude: [ :name ])
```
Comment (tas50) [inline lib/chef/mixin/properties.rb]: could we add a whitelist feature as well here?

Comment (lamont-granquist) [reply lib/chef/mixin/properties.rb]: includes is the whitelist, exclude is the blacklist

File: README.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
+
```
Comment (tas50) [inline README.md]: Is this a Tim trolling change right here?

File: lib/chef/mixin/properties.rb (hunk 1 above)
Comment (tas50) [reply lib/chef/mixin/properties.rb]: 🤦‍♂️

File: README.md (hunk 2 above)
Comment (lamont-granquist) [reply README.md]: hahahaha, no this was me fighting with git aliases and it snuck in...

**Pull request merged**